web/app/components/DepositWithdraw/blocktrades/WithdrawModalBlocktrades.jsx

```diff
@@ -198,37 +228,84 @@ class WithdrawModalBlocktrades extends React.Component {
         return "confirmation";
     }
 
-    onAccountBalance() {
+    onAccountBalance({fee,feeID,total_precision}) {
+
         if (Object.keys(this.props.account.get("balances").toJS()).includes(this.props.asset.get("id")) ) {
+            let total_minus_fee = this.props.balance.get("balance") / utils.get_asset_precision(this.props.asset.get("precision"));
+            if(this.props.asset.get("id")==feeID){
+                total_minus_fee=utils.limitByPrecision(total_minus_fee-fee*1.095,total_precision); //@#>
+                if(total_minus_fee<0){
+                    total_minus_fee=0;
+                }
+            }
+
             this.setState({
-                withdraw_amount: this.props.balance.get("balance") / utils.get_asset_precision(this.props.asset.get("precision")),
+                withdraw_amount: total_minus_fee,
+                //withdraw_amount: this.props.balance.get("balance") / utils.get_asset_precision(this.props.asset.get("precision")),
                 empty_withdraw_value: false
             });
         }
     }
 
+    setNestedRef(ref) {
+        this.nestedRef = ref;
+    }
+
+    _setTotal(asset_id, balance_id, fee, fee_asset_id) {
+
+        const gateFee = parseInt(this.props.gateFee);
+
+        let balanceObject = ChainStore.getObject(balance_id);
+        let transferAsset = ChainStore.getObject(asset_id);
+        if (balanceObject) {
+            let amount = (utils.get_asset_amount(balanceObject.get("balance"), transferAsset) - gateFee - (asset_id === fee_asset_id ? fee : 0)).toString();
+            this.setState({amount});
+        }
+    }
+
+    onFeeChanged({asset}) {
+        this.setState({feeAsset: asset, error: null});
+    }
+
+    _getAvailableAssets(state = this.state) {
+        const { from_account, from_error } = state;
+        let asset_types = [];
+        let fee_asset_types = [];
+
+        if (!(from_account && from_account.get("balances") && !from_error)) {
+            return {asset_types, fee_asset_types};
+        }
+        let account_balances = state.from_account.get("balances").toJS();
+
+        for (let key in account_balances) {
+            let asset = ChainStore.getObject(key);
+            let balanceObject = account_balances[key]?ChainStore.getObject(account_balances[key]):null;
+
+            if (balanceObject && balanceObject.get("balance") > 0) {
+                if(fee_asset_types.indexOf(key)==-1){
+                    asset_types.push(key);
+                }
+            }
+
+            if(asset&&utils.isValidPrice(asset.getIn(["options", "core_exchange_rate"]))&&parseInt(asset.getIn(["dynamic", "fee_pool"]), 10)>(this._feeBTS||2200)){
```

Comment:
So I guess the 1.095 just adds some margin in order to account for the extra fees added by the memo. I will leave it in for now and replace it once I have a proper memo fee estimation code added.